### PR TITLE
add support for "int<min,max>", "negative-int" and "numeric"

### PR DESCRIPTION
**Pull request by voku**
integer ranges:
- PHPStan: https://phpstan.org/writing-php-code/phpdoc-types#integer-ranges
- PhpStorm: https://youtrack.jetbrains.com/issue/WI-63623

### CODE REVIEW

File: src/PseudoTypes/IntegerRange.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ */
+
+namespace phpDocumentor\Reflection\PseudoTypes;
+
+use phpDocumentor\Reflection\PseudoType;
+use phpDocumentor\Reflection\Type;
+use phpDocumentor\Reflection\Types\Integer;
+
+/**
+ * Value Object representing the type 'string'.
```
Comment (orklah) [inline src/PseudoTypes/IntegerRange.php]: This should be updated :)

File: src/Types/Numeric_.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+namespace phpDocumentor\Reflection\Types;
+
+use phpDocumentor\Reflection\Type;
+
+/**
+ * Value Object representing the 'numeric' pseudo-type, which is either a numeric-string, integer or float.
+ *
+ * @psalm-immutable
+ */
+final class Numeric_ implements Type
```
Comment (orklah) [inline src/Types/Numeric_.php]: This should implemente PseudoType

**Comment by voku** [comment]
Failing test seems not related to this changes.

**Comment by jaapio** [comment]
A first thought after a quick review:

Although I like the support for the psalm types, I begin to worry about the amount of code paths we have in the type resolver. 

This is not something we should change right away, but I think before we start adding more types, I (we) should start thinking about a more robust and sustainable parser. Right now we are mixing the object creation and parsing which isn't nice. But more important, there are some less obvious bugs reported, easier to read code will help us to detect those.

**Review by jaapio — APPROVED** [review]

**Comment by jaapio** [comment]
Thanks for these additions

**Pull request merged**